tensorbay/opendataset/CCPD/loader.py

```diff
@@ -0,0 +1,233 @@
+    "U",
+    "V",
+    "W",
+    "X",
+    "Y",
+    "Z",
+    "0",
+    "1",
+    "2",
+    "3",
+    "4",
+    "5",
+    "6",
+    "7",
+    "8",
+    "9",
+    "O",
+)
+
+_CCPD_SEGMENTS = {
+    "other": ("np", "weather"),
+    "train": ("base",),
+    "val": ("base",),
+    "test": ("blur", "challenge", "db", "fn", "rotate", "tilt"),
+    "green": ("train", "val", "test"),
+}
+
+DATASET_NAME = "CCPD"
+
+
+def CCPD(path: str) -> Dataset:
+    """Dataloader of CCPD open dataset.
+
+    .. _CCPD: https://github.com/detectRecog/CCPD
+
+    The file structure should be like::
+
+        <path>
+            CCPD2019/
+                ccpd_np/
+                    1005.jpg
+                    1019.jpg
+                    ...
+                ccpd_base/
+                    00205459770115-90_85-352&516_448&547- \
+                    444&547_368&549_364&517_440&515-0_0_22_10_26_29_24-128-7.jpg
+                    00221264367816-91_91-283&519_381&553- \
+                    375&551_280&552_285&514_380&513-0_0_7_26_17_33_29-95-9.jpg
+                    ...
+                ccpd_blur/
+                ccpd_challenge/
+                ccpd_db/
+                ccpd_fn/
+                ccpd_rotate/
+                ccpd_tilt/
+                ccpd_weather/
+                LICENSE
+                README.md
+                splits/
+                    ccpd_blur.txt
+                    ccpd_challenge.txt
+                    ccpd_db.txt
+                    ccpd_fn.txt
+                    ccpd_rotate.txt
+                    ccpd_tilt.txt
+                    test.txt
+                    train.txt
+                    val.txt
+            CCPDGreen/
+                CCPD2020/
+                    ccpd_green/
+                        train/
+                        test/
+                        val/
+
+    Arguments:
+        path: The root directory of the dataset.
+
+    Returns:
+        Loaded :class: `~tensorbay.dataset.dataset.Dataset` instance.
+
+    """
+    root_path = os.path.abspath(os.path.expanduser(path))
+
+    dataset = Dataset(DATASET_NAME)
+    dataset.load_catalog(os.path.join(os.path.dirname(__file__), "catalog.json"))
+
+    for segment_head, segment_tails in _CCPD_SEGMENTS.items():
+        for segment_tail in segment_tails:
+            segment = dataset.create_segment(f"{segment_head}-{segment_tail}")
+            for image_path in _get_ccpd_image_path(root_path, segment_head, segment_tail):
+                data = Data(image_path)
+                data.label.polygon = []
+                if f"{segment_head}-{segment_tail}" != "other-np":
+                    data.label.polygon.append(_get_label(image_path))
+                segment.append(data)
+    return dataset
+
+
+def _get_license_plate(license_index: str) -> str:
+    indexes = iter(map(int, license_index.split("_")))
+    license_number = _PROVINCES[next(indexes)] + _CODES[next(indexes)]
+    for index in indexes:
+        license_number += _ADS[int(index)]
```

Comment:
python `str` is immutable, `str.__add__` creates new string. use fstring instead